flask_sqlalchemy/__init__.py

```diff
@@ -965,20 +971,21 @@ def get_binds(self, app=None):
     def _execute_for_all_tables(self, app, bind, operation, skip_tables=False):
         app = self.get_app(app)
 
-        if bind == '__all__':
-            binds = [None] + list(app.config.get('SQLALCHEMY_BINDS') or ())
-        elif isinstance(bind, string_types) or bind is None:
-            binds = [bind]
-        else:
-            binds = bind
-
-        for bind in binds:
-            extra = {}
-            if not skip_tables:
-                tables = self.get_tables_for_bind(bind)
-                extra['tables'] = tables
-            op = getattr(self.Model.metadata, operation)
-            op(bind=self.get_engine(app, bind), **extra)
+        for Base in self.bases:
+            if bind == '__all__':
```

Comment:
This `if/else` block should move outside the `for` loop because it will be the same for every iteration.